src/charm.py

```diff
@@ -46,18 +46,34 @@ def reload(self) -> bool:
             logger.debug("config reload error via %s: %s", url, str(e))
             return False
 
-    def status(self) -> Optional[dict]:
-        url = urllib.parse.urljoin(self.base_url, "/api/v2/status")
+    @staticmethod
+    def _get(url: str) -> Optional[dict]:
         try:
             response = requests.get(url)
             if response.ok and response.status_code == 200:
-                status = json.loads(response.text)
+                text = json.loads(response.text)
```

Comment:
I think that checking both `response.ok` and `response.status_code == 200` [is redundant](https://github.com/psf/requests/blob/8149e9fe54c36951290f198e90d83c8a0498289c/requests/models.py#L696)

Reply:
removed "ok"